.github/workflows/publish.yaml

```diff
@@ -0,0 +1,53 @@
+name: Build and publish an image
+
+on: # defining on which triggers this action should run
+  workflow_dispatch: # enable manual triggering of this action
+    inputs: # define the variables/inputs a user can set while triggering this action
+      input:
+        description: 'Example input'
+        required: true
+        default: 'none'
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: "maven"
+
+      - name: Build with Maven
+        run: mvn -B verify -f pom.xml -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=WARN
```

Comment:
ich würde ein `mvn install`  machen. dann funktionieren auch multimodul projekte mit internen abhängigkeiten

Reply:
Aber in einem externen Projekt soll man keine internen Abhängigkeiten haben. Ich verstehe nicht ganz wofür das gut sein soll

Reply:
hier ist ein Beispiel von einem großen Multi-Modul Projekt.  https://git.muenchen.de/ccse/unicode-tech/-/blob/master/pom.xml?ref_type=heads#L37-L50 Das sind ja dann keine internen Abhängigkeiten. Einige Module brauchen dann Sachen vom vorherigen Modul